app/src/main/java/chat/rocket/android/createchannel/ui/CreateChannelFragment.kt

```diff
@@ -0,0 +1,306 @@
+package chat.rocket.android.createchannel.ui
+
+import android.os.Bundle
+import android.view.*
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.view.ActionMode
+import androidx.core.view.isVisible
+import androidx.core.view.postDelayed
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import chat.rocket.android.R
+import chat.rocket.android.createchannel.presentation.CreateChannelPresenter
+import chat.rocket.android.createchannel.presentation.CreateChannelView
+import chat.rocket.android.main.ui.MainActivity
+import chat.rocket.android.members.adapter.MembersAdapter
+import chat.rocket.android.members.uimodel.MemberUiModel
+import chat.rocket.android.util.extensions.asObservable
+import chat.rocket.android.util.extensions.inflate
+import chat.rocket.android.util.extensions.showToast
+import chat.rocket.android.util.extensions.ui
+import chat.rocket.android.widget.DividerItemDecoration
+import chat.rocket.common.model.RoomType
+import chat.rocket.common.model.roomTypeOf
+import com.google.android.material.chip.Chip
+import dagger.android.support.AndroidSupportInjection
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import kotlinx.android.synthetic.main.fragment_create_channel.*
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class CreateChannelFragment : Fragment(), CreateChannelView, ActionMode.Callback {
+    @Inject
+    lateinit var createChannelPresenter: CreateChannelPresenter
+    private var actionMode: ActionMode? = null
+    private val adapter: MembersAdapter = MembersAdapter {
+        if (it.username != null) {
+            processSelectedMember(it.username)
+        }
+    }
+    private val compositeDisposable = CompositeDisposable()
+    private var channelType: String = RoomType.CHANNEL
+    private var isChannelReadOnly: Boolean = false
+    private var memberList = arrayListOf<String>()
+
+    companion object {
+        fun newInstance() = CreateChannelFragment()
+    }
```

Comment:
Shouldn't we put the companion object at the end of the class according to our guideline?